ccloud-monitoring/docs/consumer-scenarios/increasing-consumer-lag.rst

```diff
@@ -0,0 +1,168 @@
+.. _ccloud-monitoring-consumer-increasing-consumer-lag:
+
+Increasing Consumer Lag
+***********************
+
+Consumer lag is a tremendous performance indicator. It tells you the offset difference between the
+producer's last produced message and the consumer group's last commit. If you are unfamiliar with
+consumer groups or concepts like committing offsets, please refer to this
+`Kafka Consumer documentation <https://docs.confluent.io/platform/current/clients/consumer.html>`__.
+
+A large consumer lag, or a quickly growing lag, indicates that the consumer is not able to keep up with
+the volume of messages on a topic.
+
+This scenario will look at |ccloud| metrics from the Metrics API, `kafka-lag-exporter <https://github.com/lightbend/kafka-lag-exporter>`__ metrics, and
+client metrics from the client application’s MBean object ``kafka.consumer:type=consumer-fetch-manager-metrics,client-id=<client_id>``.
+
+Introduce failure scenario
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. By default 2 consumers and 1 producer are running. Adjust that to 1 consumer and 5 producers in order to force the condition where the consumer cannot keep up with the rate of messages being produced, which will cause an increase in consumer lag.
+   The container scaling can be done with the command below:
+
+   .. code-block:: bash
+
+      docker-compose up -d --scale consumer=1 --scale producer=5
```

Comment:
If running this without setting the `EXPORT` commands printed at the end of the start script, the output is:

```
[8:08:37] ~/git/examples/ccloud-monitoring(DEVX-2467) ✗: docker-compose up -d --scale consumer=1 --scale producer=5
WARNING: The METRICS_API_KEY variable is not set. Defaulting to a blank string.
WARNING: The METRICS_API_SECRET variable is not set. Defaulting to a blank string.
WARNING: The CLOUD_CLUSTER variable is not set. Defaulting to a blank string.
WARNING: The CONFIG_FILE variable is not set. Defaulting to a blank string.
WARNING: The BOOTSTRAP_SERVERS variable is not set. Defaulting to a blank string.
WARNING: The SASL_JAAS_CONFIG variable is not set. Defaulting to a blank string.
Recreating kafka-lag-exporter ... 
Recreating ccloud-exporter    ... 
node-exporter is up-to-date
Recreating kafka-lag-exporter                      ... done
Recreating ccloud-exporter                         ... done
Stopping and removing ccloud-monitoring_consumer_2 ... done
Recreating ccloud-monitoring_producer_1            ... done
Recreating ccloud-monitoring_consumer_1            ... done
Creating ccloud-monitoring_producer_2              ... done
Creating ccloud-monitoring_producer_3              ... done
Creating ccloud-monitoring_producer_4              ... done
Creating ccloud-monitoring_producer_5              ... done
```

- So should those export statements be executed by the user?  (if yes, that was one advantage of adding those to a `.env` file)
- Should the clients not have dependencies on the monitoring docker containers?  (so they don't get updated when scaling up/down)

Reply:
A .env file would resolve the problem when rebringing up all the containers during docker-compoes --scale but doesn't resolve why all of the containers are being evaluated to be brought up. 
Currently the only dependencies are the clients on prometheus + grafana (u right, this ain't needed). Given that, it doesn't make sense that kafka-lag-exporter, node-exporter, ccloud-exporter are recreated here. I'm digging into this.

Reply:
I removed the client dependencies in commit https://github.com/confluentinc/examples/pull/948/commits/9b6e040d6ad2773e51a2e0a49e50b944549ac902 but still see all of the containers being evaluated for "updates" when scaling. A brief look into `docker-compose up --scale` this morning didn't yield any obvious ways to circumvent this. I added an "export" of the necessary env vars to a `.env` file so that these variables are set when a --scale command is run.